File: domain/system.go
package domain

import "time"

// SystemIdentity = /system/identity/print → .name.
type SystemIdentity struct {
	Name string
}

// SystemResource = /system/resource/print.
type SystemResource struct {
	Uptime               string
	Version              string
	BuildTime            string
	FactorySoftware      string
	BoardName            string
	Platform             string
	CPU                  string
	CPUCount             int
	CPUFrequency         int64
	CPULoad              int
	FreeMemory           int64
	TotalMemory          int64
	FreeHDDSpace         int64
	TotalHDDSpace        int64
	ArchitectureName     string
	WriteSectSinceReboot int64
	BadBlocks            string
}

// SystemRouterboard = /system/routerboard/print.
type SystemRouterboard struct {
	Routerboard     bool
	Model           string
	BoardName       string
	SerialNumber    string
	FirmwareType    string
	FactoryFirmware string
	CurrentFirmware string
	UpgradeFirmware string
}

// SystemClock = /system/clock/print.
type SystemClock struct {
	Time                string
	Date                string
	TimeZoneName        string
	GMTOffset           string
	TimeZoneAutodetect  bool
	DSTActive           bool
}

// LogEntry = row /log/print.
type LogEntry struct {
	ID      string
	Time    string
	Topics  string
	Message string
	When    time.Time // best-effort, ParseTime saat list (boleh zero)
}

// Scheduler = row /system/scheduler/print.
type Scheduler struct {
	ID        string
	Name      string
	StartDate string
	StartTime string
	Interval  string
	OnEvent   string
	NextRun   string
	Policy    string
	Disabled  bool
	Comment   string
	RunCount  int
}

// Script = row /system/script/print.
//
// Mikhmon menyimpan 2 jenis: transaksi (comment="mikhmon") dan
// QuickPrint config (comment="QuickPrintMikhmon").
type Script struct {
	ID                     string
	Name                   string
	Owner                  string
	Source                 string
	Comment                string
	Policy                 string
	RunCount               int
	LastStarted            string
	DontRequirePermissions bool
}

File: website/src/types/system.ts
export interface SystemIdentity {
  name: string
  identity?: string
}

export interface SystemResource {
  uptime: string
  free_memory: number
  total_memory: number
  cpu_load: number
  cpu_frequency?: number
  version: string
  board_name: string
  architecture_name?: string
  free_hdd_space?: number
  total_hdd_space?: number
  cpu?: string
  cpu_count?: number
}

export interface Routerboard {
  routerboard: boolean
  board_name?: string
  model?: string
  serial_number?: string
  firmware_type?: string
  current_firmware?: string
  upgrade_firmware?: string
}

export interface SystemClock {
  time: string
  date: string
  timezone: string
}

export interface SystemLicense {
  software_id?: string
  n_level?: string
  features?: string
}

export interface SystemScript {
  id: string
  name: string
  source: string
  policy?: string[]
  last_started?: string
  run_count?: number
}

export interface SystemScheduler {
  id: string
  name: string
  start_date: string
  start_time: string
  interval: string
  on_event: string
  disabled?: boolean
  run_count?: number
}
